lib/OpenQA/Resource/Locks.pm

```diff
@@ -125,28 +125,32 @@ sub barrier_create {
 
 sub barrier_wait {
     my ($name, $jobid, $where, $check_dead_job) = @_;
-    return -1 unless $name && $jobid;
-    my $barrier = _get_lock($name, $jobid, $where);
-    return -1 unless $barrier;
-    my $schema    = OpenQA::Schema->singleton;
-    my $jobschema = $schema->resultset('Jobs');
-    my @jobs      = split(/,/, $barrier->locked_by // '');
-
-    do { $barrier->delete; return -1 }
-      if $check_dead_job
-      && grep { $final_states{$_} }
 
-      map { $jobschema->find($_)->result }
-      ($jobid, @jobs, map { scalar $_->id } ($barrier->owner->parents->all, $barrier->owner->children->all));
+    return -1 unless $name && $jobid;
+    return -1 unless my $barrier = _get_lock($name, $jobid, $where);
+
+    my $jobschema = OpenQA::Schema->singleton->resultset('Jobs');
+    my @jobs      = split ',', $barrier->locked_by // '';
+
+    if ($check_dead_job) {
+        my @related_ids = map { scalar $_->id } $barrier->owner->parents->all, $barrier->owner->children->all;
```

Comment:
As I said before: This should likely just be about parallel dependencies but about *all* parallel dependencies in the cluster. But I wouldn't dare to touch it, too - especially not while just doing a code cleanup.

Reply:
And especially not the day right before i go on vacation. 🤣